modules/cassandra/cassandra.go

```diff
@@ -0,0 +1,117 @@
+package cassandra
+
+import (
+	"context"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	port = nat.Port("9042/tcp")
+)
+
+// CassandraContainer represents the Cassandra container type used in the module
+type CassandraContainer struct {
+	testcontainers.Container
+}
+
+// ConnectionHost returns the host and port of the clickhouse container, using the default, native 9000 port, and
+// obtaining the host and exposed port from the container
+func (c *CassandraContainer) ConnectionHost(ctx context.Context) (string, error) {
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	port, err := c.MappedPort(ctx, port)
+	if err != nil {
+		return "", err
+	}
+
+	return host + ":" + port.Port(), nil
+}
+
+// WithConfigFile sets the YAML config file to be used for the cassandra container
+// It will also set the "configFile" parameter to the path of the config file
+// as a command line argument to the container.
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/cassandra/cassandra.yaml",
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+	}
+}
+
+// WithInitScripts sets the init cassandra queries to be run when the container starts
+func WithInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		var initScripts []testcontainers.ContainerFile
+		for _, script := range scripts {
+			cf := testcontainers.ContainerFile{
+				HostFilePath:      script,
+				ContainerFilePath: "/" + filepath.Base(script),
+				FileMode:          0o755,
+			}
+			initScripts = append(initScripts, cf)
+		}
+		req.Files = append(req.Files, initScripts...)
```

Comment:
Oh got it. So should I add them to `ContainerRequest` in `RunContainer` function like `LifecycleHooks: []testcontainers.ContainerLifecycleHooks{ ... }` or should i leave it to user to add them one by one with `testcontainers.WithStartupCommand`? Because as a user I expect from`WithInitScripts` function to run scripts its own.

Another thing, why the file name is `types_test.go` in RabbitMQ module? I thought it was a test file 😅

Reply:
> Why the file name is types_test.go in RabbitMQ module? I thought it was a test file 😅

Because I intentionally did not want to expose those types, but use them as reference for others to see how it works. Doing so allowed us not to couple with RabbitMQ implementation. Probably poor-old-man way to achieve it though! 😅 

I'd just keep the `WithInitScripts` function as you have, adding the `WithStartupCommand` internally to that method: with each new file, add its command. Does it make sense to you?

Reply:
It makes sense when knowing it, but I thought as a user, I could think that `WithInitScripts` will do all the trick and I don't need anything more, like in other modules. 

But I will stick what you said. So after I refactor it, I think I should add information about it to Readme 👍

Reply:
yes, that's what we'll achieve: the user will only know about `WithInitScripts`. We'll internally call the `WithStartupCommand` for every file.

Please check this code snippet locally before commiting:

```golang

type initScript struct {
	File     string
}

func (i initScript) AsCommand() []string {
	if strings.HasSuffix(i.File, ".cql") {
		container.Exec(ctx, []string{"cqlsh", "-f", i.File})
	} else if strings.HasSuffix(i.File, ".sh") {
		container.Exec(ctx, []string{"/bin/sh", i.File})
	}

	// what do we return here if the file extension is incorrect? an empty array, an incorrect file to raise an error later on?
}

func WithInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
	return func(req *testcontainers.GenericContainerRequest) {
		var initScripts []testcontainers.ContainerFile
		for _, script := range scripts {
			cf := testcontainers.ContainerFile{
				HostFilePath:      script,
				ContainerFilePath: "/" + filepath.Base(script),
				FileMode:          0o755,
			}
			initScripts = append(initScripts, cf)

			testcontainers.WithStartupCommand(initScript{File: "/" + filepath.Base(script)})
		}
```

Reply:
Ah alright. This makes more sense to me. I initially thought like this, but when I saw it in RabbitMQ tests, I thought the user would add `WithStartupCommand`.
It's more clear now. Thank you

Reply:
Yeah, sure I can understand that :D textual communications could be hard! Hopefully we got to it very fast. Thanks for your patience!